### ISSUE TITLE
RDS arn missing region

### ISSUE THREAD

**Issue by jdubs** [capitalone/cloud-custodian#801]
There's an issue tagging RDS instances when region isn't passed in on the command line. In https://github.com/capitalone/cloud-custodian/blob/master/c7n/utils.py#L244 if the region isn't set, it will generate invalid RDS ARNs, for instance. Passing in `-r us-east-1` will generate a syntactically correct arn, `arn:aws:rds:us-east-1:123123123:db:trenholmtest` with out region, and without the  ARN. `arn:aws:rds::123123123:db:trenholmtest`

In resources/s3.py it uses us-east-1 as the default if it's not specified. Should RDS be the same way or is this by design?

example policy
```
policies:
  - name: rds-owner-tag
    resource: rds
    filters:
      - "tag:OwnerName": absent
    actions:
      - type: tag
        key: OwnerName
        value: OwnerName
```
After adding some debug
```
arn:aws:rds::123123123:db:trenholmtest
An error occurred (AccessDenied) when calling the ListTagsForResource operation: Unknown
Traceback (most recent call last):
  File "/Users/jtrenholm/work/github-cloud-custodian/cloud-custodian/c7n/resources/rds.py", line 151, in process_tags
    tag_list = retry(client.list_tags_for_resource, ResourceName=arn)['TagList']
  File "/Users/jtrenholm/work/github-cloud-custodian/cloud-custodian/c7n/utils.py", line 288, in _retry
    return func(*args, **kw)
  File "/Users/jtrenholm/work/github-cloud-custodian/cloud-custodian/lib/python2.7/site-packages/botocore/client.py", line 251, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/jtrenholm/work/github-cloud-custodian/cloud-custodian/lib/python2.7/site-packages/botocore/client.py", line 537, in _make_api_call
    raise ClientError(parsed_response, operation_name)
ClientError: An error occurred (AccessDenied) when calling the ListTagsForResource operation: Unknown
```

**Comment by kapilt** [comment]
we should probably be falling back to the session's region which will take into account metadata server for an instance.

**Comment by kapilt** [comment]
its a bit unclear how to tackle this cleanly.. and it effects any resource that requires arns for tags.

**Comment by kapilt** [comment]
#835 is an attempt to resolve

**Comment by kapilt** [comment]
This should be resolved in 0.8.23.0 as region won't ever be null to custodian, it will always match the aws sdk.
